src/sync.d

```diff
@@ -1132,16 +1137,19 @@ final class SyncEngine
 		
 		if (!dryRun) {
 			ulong fileSize = 0;
-			if ( (hasFileSize(fileDetails)) && (fileDetails.object()) ) {
+			string OneDriveFileHash;
+			if ( (hasFileSize(fileDetails)) && (hasQuickXorHash(fileDetails)) && (fileDetails.object()) ) {
+				// fileDetails is a valid JSON object with the elements we need
 				// Set the file size from the returned data
 				fileSize = fileDetails["size"].integer;
+				OneDriveFileHash = fileDetails["file"]["hashes"]["quickXorHash"].str;
 			} else {
 				// Issue #540 handling
 				log.vdebug("ERROR: onedrive.getFileDetails call returned a OneDriveException error");
```

Comment:
Probably my insufficiency to know the protocol, but will there **always** be a hash (`hasQuickXorHash`) be available in a proper response? Otherwise we would get errors and redownloading over and over again.

Reply:
@norbusan 
MS provide SHA1 and QuickXor Hash in Personal Accounts and only QuickXor in Business / O365 - and this always has to be part of the 'file' attribute as part of the spec